framework/io/obj_io.cc

```diff
@@ -0,0 +1,392 @@
+
+      auto cell = new UnpartitionedMesh::LightWeightCell(CellType::POLYGON, sub_type);
+      cell->material_id = material_id;
+
+      // Populate vertex-ids
+      for (size_t k = 1; k <= number_of_verts; k++)
+      {
+        // Extract sub word
+        beg_of_word = file_line.find_first_not_of(delimiter, end_of_word);
+        end_of_word = file_line.find(delimiter, beg_of_word);
+        sub_word = file_line.substr(beg_of_word, end_of_word - beg_of_word);
+
+        // Extract locations of hyphens
+        size_t first_dash = sub_word.find('/');
+        size_t last_dash = sub_word.find_last_of('/');
+
+        // Extract the words ass. vertex and normal
+        std::string vert_word = sub_word.substr(0, first_dash - 0);
+        std::string norm_word = sub_word.substr(last_dash + 1, sub_word.length() - last_dash - 1);
+
+        // Convert word to number (Vertex)
+        try
+        {
+          int numValue = std::stoi(vert_word);
+          cell->vertex_ids.push_back(numValue - 1);
+        }
+        catch (const std::invalid_argument& ia)
+        {
+          log.Log0Warning() << "Failed converting work to number in line " << file_line
+                            << std::endl;
+        }
+
+        // Stop word extraction on line-end
+        if (end_of_word == std::string::npos)
+        {
+          break;
+        }
+      }
+
+      // Build faces
+      const size_t num_verts = cell->vertex_ids.size();
+      for (uint64_t v = 0; v < num_verts; ++v)
+      {
+        UnpartitionedMesh::LightWeightFace face;
+
+        face.vertex_ids.resize(2);
+        face.vertex_ids[0] = cell->vertex_ids[v];
+        face.vertex_ids[1] = (v < (num_verts - 1)) ? cell->vertex_ids[v + 1] : cell->vertex_ids[0];
+
+        cell->faces.push_back(std::move(face));
+      }
+
+      if (block_data.empty())
+        throw std::logic_error(fname + ": Could not add cell to block-data. "
+                                       "This normally indicates that the file does not have the "
+                                       "\"o Object Name\" entry.");
+
+      block_data.back().cells.push_back(cell);
+    } // if (first_word == "f")
+
+    // Keyword "l" for edge
+    if (first_word == "l")
+    {
+      Edge edge;
+      for (int k = 1; k <= 2; ++k)
+      {
+        // Extract sub word
+        beg_of_word = file_line.find_first_not_of(delimiter, end_of_word);
+        end_of_word = file_line.find(delimiter, beg_of_word);
+        sub_word = file_line.substr(beg_of_word, end_of_word - beg_of_word);
+
+        // Convert word to number
+        try
+        {
+          int vertex_id = std::stoi(sub_word);
+          if (k == 1)
+            edge.first = vertex_id - 1;
+          if (k == 2)
+            edge.second = vertex_id - 1;
+        }
+
+        // Catch conversion error
+        catch (const std::invalid_argument& ia)
+        {
+          log.Log0Warning() << "Failed to text to integer in line " << file_line << std::endl;
```

Comment:
I think we're missing a `convert` here. Maybe `Failed to convert text to integer in line`.